js/kucoinfutures.js

```diff
@@ -0,0 +1,1287 @@
+            side = 'sell';
+        }
+        const notional = Precise.stringAbs (this.safeString (position, 'posCost'));
+        const initialMargin = this.safeString (position, 'posMargin');
+        const initialMarginPercentage = Precise.stringDiv (initialMargin, notional);
+        const leverage = Precise.stringDiv ('1', initialMarginPercentage);  // TODO: Not quite right
+        // const marginRatio = Precise.stringDiv (maintenanceRate, collateral);
+        const unrealisedPnl = this.safeString (position, 'unrealisedPnl');
+        return {
+            'info': position,
+            'symbol': this.safeString (market, 'symbol'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'initialMargin': this.parseNumber (initialMargin),
+            'initialMarginPercentage': this.parseNumber (initialMarginPercentage),
+            'maintenanceMargin': this.safeNumber (position, 'maintMargin'),
+            'maintenanceMarginPercentage': this.safeString (position, 'maintMarginReq'),
+            'entryPrice': this.safeNumber (position, 'avgEntryPrice'),
+            'notional': this.parseNumber (notional),
+            'leverage': this.parseNumber (leverage),
+            'unrealizedPnl': this.parseNumber (unrealisedPnl),
+            'contracts': this.parseNumber (Precise.stringAbs (size)),
+            'contractSize': this.safeNumber (market, 'contractSize'),
+            //     realisedPnl: position['realised_pnl'],
+            'marginRatio': undefined,
+            'liquidationPrice': this.safeNumber (position, 'liquidationPrice'),
+            'markPrice': this.safeNumber (position, 'markPrice'),
+            'collateral': this.safeNumber (position, 'posInit'),
+            'marginType': undefined,
+            'side': side,
+            'percentage': this.parseNumber (Precise.stringDiv (unrealisedPnl, initialMargin)),
+        };
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const uppercaseType = type.toUpperCase ();
+        const market = this.market (symbol);
+        // required param, cannot be used twice
+        const clientOrderId = this.safeString2 (params, 'clientOid', 'clientOrderId', this.uuid ());
+        params = this.omit (params, [ 'clientOid', 'clientOrderId' ]);
+        const leverage = this.safeNumber (params, 'leverage');
+        if (!leverage) {
+            throw new ArgumentsRequired (this.id + ' createOrder requires params.leverage');
+        }
+        if (amount < 1) {
+            throw new InvalidOrder ('Minimum contract order size using ' + this.id + ' is 1');
+        }
+        const stop = this.safeString (params, 'stop');
+        const stopPrice = this.safeNumber (params, 'stopPrice');
+        if (stop) {
+            const stopPriceType = this.safeString (params, 'stopPriceType');
+            if (!stopPriceType || !stopPrice || !stop) {
```

Comment:
instead of this conditional you could refactor it to this:

`if (!stopPriceType || !stopPrice)`